### PR TITLE
fix: Skip non-init fields in dataclasses.

### LINKED ISSUE #181: bug: Cappa tries to pass dataclass classvar as keyword argument
```python
from __future__ import annotations

import sys
import cappa
from dataclasses import dataclass
from typing import ClassVar


@cappa.command(name="command")
@dataclass(kw_only=True)
class Command:
    _CONFIG: ClassVar[str] = "config.toml"

    arg: int = 0

    def __call__(self):
        print("Command called")



def main(args: list[str] | None = None) -> int:
    return cappa.invoke(Command, argv=args)


if __name__ == "__main__":
    sys.exit(main())
```

```pytb
Traceback (most recent call last):
  File "/media/data/dev/insiders-project/repro.py", line 26, in <module>
    sys.exit(main())
             ^^^^^^
  File "/media/data/dev/insiders-project/repro.py", line 22, in main
    return cappa.invoke(Command, argv=args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/insiders-project/.venv/lib/python3.12/site-packages/cappa/base.py", line 131, in invoke
    command, parsed_command, instance, concrete_output = parse_command(
                                                         ^^^^^^^^^^^^^^
  File "/media/data/dev/insiders-project/.venv/lib/python3.12/site-packages/cappa/base.py", line 247, in parse_command
    command, parsed_command, instance = Command.parse_command(
                                        ^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/insiders-project/.venv/lib/python3.12/site-packages/cappa/command.py", line 188, in parse_command
    result = command.map_result(command, prog, parsed_args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/data/dev/insiders-project/.venv/lib/python3.12/site-packages/cappa/command.py", line 239, in map_result
    return command.cmd_cls(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Command.__init__() got an unexpected keyword argument '_CONFIG'
```

### PR DESCRIPTION
**Pull request by pawamoy**
Closes #181.

### CODE REVIEW

**Comment by pawamoy** [comment]
The type-checking error could be fixed by changing the annotation of `Field.metadata` to `Mapping`. Or we can type-ignore it, let me know.

**Comment by DanCardin** [comment]
i'd probably dict it personally, for making a type that can be consistent across the various types.

File: src/cappa/class_inspect.py [hunk 1]
```diff
@@ -31,23 +33,25 @@ class Field:
     annotation: type
     default: typing.Any | EmptyType = Empty
     default_factory: typing.Any | EmptyType = Empty
-    metadata: dict = dataclasses.field(default_factory=dict)
+    metadata: Mapping = dataclasses.field(default_factory=dict)
```
Comment (pawamoy) [inline src/cappa/class_inspect.py:36]: Oops, also committed this change, let me revert.

Comment (pawamoy) [reply src/cappa/class_inspect.py:36]:
```suggestion
    metadata: dict = dataclasses.field(default_factory=dict)
```

File: src/cappa/class_inspect.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
 from cappa.typing import T, find_annotations
 
 if typing.TYPE_CHECKING:
+    from collections.abc import Mapping
+
```
Comment (pawamoy) [inline src/cappa/class_inspect.py:17]:
```suggestion
```

**Pull request merged**